t/harness

```diff
@@ -114,133 +240,50 @@ if (@ARGV) {
-	    # (in the right order). Not sure if this is a bug or a feature.
-	    # Whatever, *we* are only interested in the ones that are in @tests
-	    my %seen;
-	    @seen{@tests} = ();
-	    @tests = grep {exists $seen{$_} } $state->get_tests(0, @tests);
-	}
-	@tests = (@seq, @tests);
-	push @seq, $next;
-
-	push @last,
-	    _tests_from_manifest($Config{extensions}, $Config{known_extensions});
-	my %times;
-	if ($state) {
-	    # Where known, collate the elapsed times by test name
-	    foreach ($state->results->tests()) {
-		$times{$_->name} = $_->elapsed();
-	    }
-	}
-
-	my %dir;
-	my %total_time;
-        my %serials;
-        my %all_dirs;
-
-        # Preprocess the list of tests
-	for (@last) {
-	    if ($^O eq 'MSWin32') {
-		s,\\,/,g; # canonicalize path
-	    };
-
-            # Keep a list of the distinct directory names, and another list of
-            # those which contain a file whose name begins with a 0
-            if ( m! \A (?: \.\. / )?
-                                ( .*? )         # $1 is the directory path name
-                            /
-                                ( [^/]* \.t )   # $2 is the .t name
-                    \z !x)
-            {
-                my $path = $1;
-
-                $all_dirs{$path} = 1;
-                $serials{$path} = 1 if $2 =~ / \A 0 /x;
-            }
+        push @$which, 'japh' if $torture;
+        push @$which, 'win32' if $^O eq 'MSWin32';
+        push @$which, 'benchmark' if $ENV{PERL_BENCHMARK};
+        push @$which, 'bigmem' if $ENV{PERL_TEST_MEMORY};
+
+        if (@next) {
+            # Hopefully TAP::Parser::Scheduler will support this syntax soon.
+            my $next = { par => '{' . join (',', @next) . '}/*.t' };
+            push @tests, _extract_tests ($next);
+            push @seq, _compute_tests_and_ordering(\@tests)->@*;
         }
 
-        # We assume that the reason a test file's name begins with a 0 is to
-        # order its execution among the tests in its directory.  Hence, a
-        # directory containing such files should be tested in serial order.
-        #
-        # Add exceptions to the above rule
-        for (qw(cpan/IO-Zlib/t ext/File-Find/t)) {
-            $serials{$_} = 1;
-        }
+        my $last = { par => '{' . join (',', @last) . '}/*.t' };
+        @last = _extract_tests ($last);
 
-        my @nonexistent_serials = grep { not exists $all_dirs{$_} } keys %serials;
-        if (@nonexistent_serials) {
-            die "These directories to be run serially don't exist."
-              . "  Check your spelling:\n" . join "\n", @nonexistent_serials;
+        # This is a bit of a game, because we only want to sort these tests in
+        # speed order. base/*.t wants to run first, and ext,lib etc last and in
+        # MANIFEST order
+        push @last,
+            _tests_from_manifest($Config{extensions}, $Config{known_extensions});
+        push @tests, @last;
+
+        push @tests, @last;
```

Comment:
This line is duplicated, and is the cause of the test fails.

Reply:
a rebasing error